src/Microsoft.DocAsCode.Build.Engine/Incrementals/IncrementalBuildContext.cs

```diff
@@ -645,7 +676,8 @@ private bool GetCanVersionIncremental(IncrementalStatus buildInfoIncrementalStat
                 Logger.LogVerbose(message);
                 return false;
             }
-            if (CurrentBuildVersionInfo.ConfigHash != LastBuildVersionInfo.ConfigHash)
+            if (CurrentBuildVersionInfo.ConfigHash != LastBuildVersionInfo.ConfigHash
+                || CurrentBuildVersionInfo.FileMetadataHash != LastBuildVersionInfo.FileMetadataHash) // TODO: move to file level check
             {
                 string message = "Cannot build incrementally because config changed.";
```

Comment:
>Cannot build incrementally because config changed. [](start = 34, length = 50)

it is not related to this change, but is it possible to include what part of config is changed in the information?

Reply:
Yes, we can add the whole config into cache. When config is different, we can log a message based on property wise comparison.